chimney/.jvm/src/test/scala/io/scalaland/chimney/WixSpec.scala

```diff
@@ -308,16 +324,34 @@ object WixSpec extends TestSuite {
           .withFieldConst(_.otherValue, "default")
           .buildTransformer
 
-        EntityDTO(None).transformInto[Entity] ==> Entity(SdlMissingIdPlaceholder, "default")
+        EntityDTO(None).transformInto[Entity] ==> Entity(SdlMissingIdPlaceholderString, "default")
+      }
+
+      "support UUID placeholder" - {
+        case class Entity(@id(UUIDCompatible, IdGeneration.Auto) id: UUID)
+
+        implicit val t: Transformer[String, UUID] = str => UUID.fromString(str)
+
+        EntityDTO(None).transformInto[Entity] ==> Entity(UUID.fromString(SdlMissingIdPlaceholderUUID))
```

Comment:
Instead of `UUID.fromString` why not define `SdlMissingIdPlaceholderUUID` as `UUID(0,0)` ?

Reply:
great question! :) 
it was my initial approach, however I had to resort to the string
the [placeholder](https://github.com/wix-incubator/chimney/blob/38ffb293523bc180e188c7a00ae745eb43dbb61f/chimney/src/main/scala/io/scalaland/chimney/internal/macros/TransformerMacros.scala#L309) has to be a String (since the source is string as well).

The alternative would be to write something like this:
`$srcPrefixTree.map(UUID.fromString).getOrElse($placeholder)` and it will work, but it breaks [this test](https://github.com/wix-incubator/chimney/pull/26/files#diff-f6c2c389b2e3131b0f1b4b05611b76d0390cfcd16215b0005da6635007169ecaR348) (we want to look for an implicit String=>UUID transformer, not perform transformation in-palace).

We could've looked it up using `resolveImplicitTransformer()` invocation, but that will complicate the code since:
* it's called down the line the second time, so we need to adjust [From, To ](https://github.com/wix-incubator/chimney/blob/38ffb293523bc180e188c7a00ae745eb43dbb61f/chimney/src/main/scala/io/scalaland/chimney/internal/macros/TransformerMacros.scala#L310)
* we need to define an implicit [`Liftable`](https://docs.scala-lang.org/overviews/quasiquotes/lifting.html) for `UUID`
* we need an overload for `expandSourceStringWrappedInOptionWithSdlIdPlaceholder` with `placeholder: UUID` 

I've decided to go with String since it results in a more straightforward code